backend/dataall/modules/s3_datasets/api/storage_location/resolvers.py

```diff
@@ -47,6 +50,13 @@ def resolve_dataset(context, source: DatasetStorageLocation, **kwargs):
     if not source:
         return None
     with context.engine.scoped_session() as session:
+        ResourcePolicyService.check_user_resource_permission(
+            session=session,
+            username=get_context().username,
+            groups=get_context().groups,
+            resource_uri=source.datasetUri,
+            permission_name=GET_DATASET,
+        )
```

Comment:
As an alternative we could create a find_dataset method in DatasetService similar to `EnvironmentService.find_environment_by_uri(uri)` and that way avoid application logic and checks happening in the resolver